exercises/exercise_6/README.md

```diff
@@ -1 +1,519 @@
+- Metadata: application name, version, description, maintainers
+- List of invocation images
+- List of component images (compose services in our case)
+- Custom actions: every bundle must implement the 3 mandatory actions **install**/**upgrade**/**uninstall**
+- Parameters for these actions
+- Credentials
+
+Here is an example of a simple `bundle.json`:
+
+```json
+{
+	"name": "simple",
+	"version": "1.0.0",
+	"description": "a very simple application",
+	"maintainers": [
+		{
+			"name": "DAP Workshop",
+			"email": "dapworkshop@docker.com"
+		}
+	],
+	"invocationImages": [
+		{
+			"imageType": "docker",
+			"image": "simple:1.0.0-invoc"
+		}
+	],
+	"images": {
+		"api": {
+			"imageType": "docker",
+			"image": "simple-component:1.0.0",
+			"description": "simple component"
+		}
+	},
+	"actions": {
+		"io.cnab.status": {}
+	},
+	"parameters": {
+		"simple_component_port": {
+			"type": "string",
+			"defaultValue": "8080",
+			"destination": {
+				"env": "SIMPLE_COMPONENT_PORT"
+			}
+		}
+	},
+	"credentials": {}
+}
+```
+
+### Invocation image
+
+**NOTE** During this exercise, we will only focus on Docker base invocation images.
+
+The second artifact is the invocation image, which is a simple Docker image, able to install the application. It's not the application itself, see it as the installer binary.
+
+It must:
+- contain a simple executable file at the path **`/cnab/app/run`**
+- understand and react to actions declared in the `bundle.json`
+
+Some parts are injected dynamically in the container by the runtime, like the credentials or the parameters.
+
+When the `CNAB runtime` executes an action, it will follow these steps:
+1. Parse and validate the bundle.json file
+1. Resolve and validate (type, allowed values, required) the parameters
+1. Create a container using the specified invocation image
+1. Create environment variable with the installation name (`CNAB_INSTALLATION_NAME`), bundle name (`CNAB_BUNDLE_NAME`), bundle version (`CNAB_BUNDLE_VERSION`), and finally the action to execute (`CNAB_ACTION`)
+1. Either create environment variables or mount files for parameters
+1. Mount with `tmpfs`the credentials to the specified paths
+1. Run the container and execute `/cnab/app/run`
+1. Display the container's stdout/stderr
+1. Create or Update a claim to store the action result (`SUCCESS` or `FAILURE`), if it is not stateless
+
+## Create a CNAB Bundle from scratch
+
+Let's put our hands on what we just saw an create our first CNAB!
+
+1. Create a new directory called `simple`
+1. In it, copy/paste the previous bundle example in a new file `bundle.json`
+1. Still in `simple`, create the following hierarchy:  
+
+    ```bash
+    $ tree simple
+    simple
+    ├── bundle.json
+    └── cnab
+        ├── app
+        │   └── run
+        └── build
+            └── Dockerfile
+
+    3 directories, 3 files
+    ```
+
+**EXERCISE**: Write the `simple/cnab/build/Dockerfile` to build the invocation image
```

Comment:
To keep the exercises consistent with each other, I think this should be a heading. Probably a level 3, as we are still under "Create a CNAB from scratch"

```suggestion
### Creating the Dockerfile for the invocation image
``